dali/imgcodec/image_format.cc

```diff
@@ -67,5 +67,19 @@ span<ImageFormat* const> ImageFormatRegistry::Formats() const {
   return make_cspan(format_ptrs_);
 }
 
+size_t ImageParser::ReadHeader(ImageSource *encoded, uint8_t *buffer, size_t n) const {
+  if (encoded->Kind() == InputKind::HostMemory) {
+    if (encoded->Size() < n)
+      n = encoded->Size();
+    std::copy_n(encoded->RawData<char>(), n, buffer);
+  } else {
+    auto stream = encoded->Open();
+    if (stream->Size() < n)
+      n = stream->Size();
+    stream->ReadBytes(buffer, n);
```

Comment:
```suggestion
    return stream->Read(buffer, n);
```
This will perform a partial Read - there's no need to reinvent it.

Reply:
You're right, done